civictechprojects/models.py

```diff
@@ -215,10 +215,17 @@ def recache(self):
     def generate_full_text(self):
         base_json = self.hydrate_to_json()
         # Don't cache external entities because they take up space and aren't useful in project search
-        del base_json['project_volunteers']
-        del base_json['project_events']
-        del base_json['project_groups']
-        del base_json['project_commits']
+        omit_fields = ['project_volunteers', 'project_owners', 'project_events', 'project_groups', 'project_commits']
+        # Don't cache files because they contain noise without adequate signal
+        omit_fields += ['project_thumbnail', 'project_files']
+        # Don't cache boolean fields
+        omit_fields += ['project_claimed', 'project_approved']
+        # Don't cache numeric fields
+        omit_fields += ['project_id', 'project_creator', 'project_latitude', 'project_longitude']
+        # Don't cache date fields
+        omit_fields += ['project_date_modified']
+        for field in omit_fields:
+            base_json.pop(field, None)
```

Comment:
Are there performance concerns with python's pop() here, potentially down the line on a larger set of data to work with? 

It clearly works - I searched for the string democracylab on two environments and got 117 results (including incorrect results) or 6 (correctly) and if I understand how this caching function works it's not likely to be impacting the user experience except in significantly better full-text search results for a few key phrases.

Reply:
I'm not overly concerned with performance in this bit of code, as it only runs when objects are updated, as part of an operation that includes a number of database reads.